src/main/cpp/src/timezones.cu

```diff
@@ -118,6 +120,183 @@ auto convert_timestamp_tz(column_view const& input,
   return results;
 }
 
+template <typename timestamp_type>
+struct time_add_functor {
+  using duration_type = typename timestamp_type::duration;
+
+  lists_column_device_view const transitions;
+
+  size_type const tz_index;
+
+  int64_t const duration_scalar;
+
+  __device__ inline timestamp_type plus_with_tz(timestamp_type const& timestamp,
+                                                int64_t const& duration) const
+  {
+    if (duration == 0L) { return timestamp; }
+
+    auto const utc_instants = transitions.child().child(0);
+    auto const tz_instants  = transitions.child().child(1);
+    auto const utc_offsets  = transitions.child().child(2);
+
+    auto const epoch_seconds_utc = static_cast<int64_t>(
```

Comment:
Why do we need to cast the result to int64_t? What is the type that this returns? if duration for seconds in not an `int64_t` or larger already we risk overflowing the result.